build/dsc/psmodule.rb

```diff
@@ -25,7 +25,7 @@ def attributes
             utf8_encoded_content.lines.each do |line|
               dos2unix(line)
               matches = regex.match(line)
-              attrs[matches[1].strip] = matches[2] if matches
+              attrs[matches[1].strip.downcase] = matches[2] if matches
```

Comment:
Only `name` and `version` are used, and they are used in [dsc_type.rb.erb#L79-L81](https://github.com/puppetlabs/puppetlabs-dsc/blob/master/build/dsc/templates/dsc_type.rb.erb#L79-L81) and [dsc_type_spec.rb.erb#L256](https://github.com/puppetlabs/puppetlabs-dsc/blob/master/build/dsc/templates/dsc_type_spec.rb.erb#L256). 

If there were other usages that could be broken by case, it would have already blown up when building the types. I will make it private to be explicit on scope.

Reply:
Looks like `attributes` does appear in the builder specs at https://github.com/puppetlabs/puppetlabs-dsc/blob/c4231faa2dffa16b8b9c64a846a054e77d09851c/spec/unit/psmodule_spec.rb - though it has no bearing here. It does prevent us from adding `private` though.

It wouldn't have necessarily blown up if there were guards for the attribute reads like this:

```ruby
foo = attributes['Name'] if attributes.key?('Name')
```
Fortunately, looks to not be the case.

In the future if you do a before / after diff for generated types in the PR comments, then it makes it easier to do a quick merge.